extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/KafkaRecordSupplier.java

```diff
@@ -207,7 +210,40 @@ public static void addConsumerPropertiesFromConfig(
     ClassLoader currCtxCl = Thread.currentThread().getContextClassLoader();
     try {
       Thread.currentThread().setContextClassLoader(getClass().getClassLoader());
-      return new KafkaConsumer<>(props, new ByteArrayDeserializer(), new ByteArrayDeserializer());
+      Deserializer keyDeserializerObject;
+      Deserializer valueDeserializerObject;
+  
+      try {
```

Comment:
Could you consider extracting this to a helper method -- maybe something called like this:

```java
Deserializer deserializer = getDeserializer(props, "key.deserializer")
```

That way, the (mostly) identical code block wouldn't need to be duplicated.

Reply:
Very good idea.  I should have done that to begin with.  I have implemented this.